src/hooks/routing/useRouterTrade.ts

```diff
@@ -41,50 +42,49 @@ export function useRouterTrade(
-  const { isError, data, currentData, fulfilledTimeStamp } = useGetQuoteQueryState(queryArgs)
-  const isValidBlock = useIsValidBlock(Number(data?.blockNumber))
-  const isSyncing = currentData !== data || !isValidBlock
+  const { isError, isFetching, data, currentData, fulfilledTimeStamp } = useGetQuoteQueryState(queryArgs)
 
-  // PRICE fetching is informational and costly, so it's done less frequently.
-  const pollingInterval = routerPreference === RouterPreference.PRICE ? ms`2m` : ms`15s`
+  // An already-fetched value should be refetched if it is older than the pollingInterval.
+  // Without explicit refetch, it would not be refetched until another pollingInterval has elapsed.
   const [trigger] = useLazyGetQuoteQuery({ pollingInterval })
   const request = useCallback(() => {
     const { refetch } = trigger(queryArgs, /*preferCacheValue=*/ true)
-    // An already-fetched value should be refetched if it is older than the pollingInterval.
-    // Without explicit refetch, it would not be refetched until another pollingInterval has elapsed.
-    if (fulfilledTimeStamp && Date.now() - fulfilledTimeStamp > pollingInterval) {
+    if (!isFetching && fulfilledTimeStamp && Date.now() - fulfilledTimeStamp > pollingInterval) {
       refetch()
     }
-  }, [fulfilledTimeStamp, pollingInterval, queryArgs, trigger])
+  }, [fulfilledTimeStamp, isFetching, pollingInterval, queryArgs, trigger])
   useTimeout(request, 200)
 
-  const [currencyIn, currencyOut] = isExactInput(tradeType)
-    ? [amountSpecified?.currency, otherCurrency]
-    : [otherCurrency, amountSpecified?.currency]
+  const result = useMemo(() => (typeof data === 'object' ? data : undefined), [data])
   const trade = useMemo(() => {
-    const routes = computeRoutes(currencyIn, currencyOut, tradeType, data)
+    const [currencyIn, currencyOut] = isExactInput(tradeType)
+      ? [amountSpecified?.currency, otherCurrency]
+      : [otherCurrency, amountSpecified?.currency]
+    const routes = computeRoutes(currencyIn, currencyOut, tradeType, result)
     if (!routes || routes.length === 0) return
     try {
       return transformRoutesToTrade(routes, tradeType)
     } catch (e: unknown) {
       console.debug('transformRoutesToTrade failed: ', e)
       return
     }
-  }, [currencyIn, currencyOut, data, tradeType])
-  const gasUseEstimateUSD = useStablecoinAmountFromFiatValue(data?.gasUseEstimateUSD)
+  }, [amountSpecified?.currency, otherCurrency, result, tradeType])
+  const isValidBlock = useIsValidBlock(Number(result?.blockNumber))
+  const isSyncing = currentData !== data || !isValidBlock
+  const gasUseEstimateUSD = useStablecoinAmountFromFiatValue(result?.gasUseEstimateUSD)
 
   return useMemo(() => {
-    if (!currencyIn || !currencyOut) return TRADE_INVALID
-    if (!trade && !isError) return TRADE_LOADING
+    if (queryArgs === skipToken) return TRADE_INVALID
+    if (data === NO_ROUTE) return TRADE_NOT_FOUND
```

Comment:
This would otherwise be treated as `TRADE_LOADING`, so I created a new return value (`NO_ROUTE`) to signify this state. `NO_ROUTE` is a string so that redux can serialize it - redux cautions against using a Symbol.